packages/mobile/src/components/track-details-tile/TrackDetailsTile.tsx

```diff
@@ -73,23 +83,70 @@ type TrackDetailsTileProps = {
 
 export const TrackDetailsTile = ({ trackId }: TrackDetailsTileProps) => {
   const styles = useStyles()
-  const accentBlue = useColor('accentBlue')
+  const { accentBlue, specialLightGreen1 } = useThemeColors()
   const track = useSelector((state) => getTrack(state, { id: trackId }))
   const owner = useSelector((state) => getUser(state, { id: track?.owner_id }))
   const isCollectibleGated = isPremiumContentCollectibleGated(
     track?.premium_conditions
   )
+  const isUSDCPurchaseGated =
+    useIsUSDCEnabled() &&
+    isPremiumContentUSDCPurchaseGated(track?.premium_conditions)
   const { doesUserHaveAccess } = usePremiumContentAccess(track)
 
   const dogEarType = getDogEarType({
     doesUserHaveAccess,
     premiumConditions: track?.premium_conditions
   })
 
+  const type = isUSDCPurchaseGated
+    ? PremiumContentType.USDC_PURCHASE
+    : isCollectibleGated
+    ? PremiumContentType.COLLECTIBLE_GATED
+    : PremiumContentType.SPECIAL_ACCESS
+
+  const headerAttributes: Record<
```

Comment:
had to explicitly type this in order to get colorString as a keyof ThemeColors type. Open to suggestions!

Reply:
💯  This is close to a pattern I regularly follow for mapping an enum or limited set of choices to a set of values/variants of a component. 

My only suggestion might be that if you want to make sure this always maps every enum value, don't use a `Record`, but a [mapped type](https://www.typescriptlang.org/docs/handbook/2/mapped-types.html) :
```typescript
// defined outside the component
type HeaderAttributesMapping = { [k in PremiumContentType]: {
  message: string
  icon: ComponentType<SvgProps>
  color: string
  colorString: keyof ThemeColors
}}

// Used in the assignment
const headerAttributes: HeaderAttributesMapping = ...
```

Doing it that way _requires_ a value for every key in the `PremiumContentType` enum. So if we change that enum and don't change this definition, we will get a compiler error. It also has the benefit of the compiler _knowing_ that the value of headerAttributes will be defined, since all of the possible keys in this mapping have corresponding values.

Reply:
ooo nice ty!